client/client.go

```diff
@@ -1043,6 +1086,24 @@ func (c *Client) setupDrivers() error {
 	return nil
 }
 
+// updateNodeFromFingerprint updates the node with the result of
+// fingerprinting the node from the diff that was created
+func (c *Client) updateNodeFromFingerprint(response *cstructs.FingerprintResponse) {
+	c.configLock.Lock()
+	defer c.configLock.Unlock()
+	for name, val := range response.Attributes {
+		c.config.Node.Attributes[name] = val
+	}
+
+	// update node links and resources from the diff created from
+	// fingerprinting
+	for name, val := range response.Links {
+		c.config.Node.Links[name] = val
+	}
+
+	c.config.Node.Resources.Merge(response.Resources)
```

Comment:
I am fine with this strategy for now but this doesn't let a fingerprinter report a non-zero value and then set it to zero. The long term solution is that we pass in a different struct type that has all the fields as a pointer and as long as the fingerprinter sets the pointer we merge the value

Reply:
I'll add that as a tech task to complete after this refactor.